runtime/interpreter/value.go

```diff
@@ -1944,6 +1948,7 @@ func (v *ArrayValue) Transfer(
 	remove bool,
 	storable atree.Storable,
 ) Value {
+	interpreter.ReportComputation(common.ComputationKindTransferArrayValue, uint(v.Count()))
```

Comment:
`Count` uses `v.array`, which may be nil if the array is a resource and got invalidated. Move this line after the if statement below

Reply:
Maybe also double-check the other functions where `ReportComputation` is called and ensure it is after `checkInvalidatedResourceUse`

Reply:
updated at 7f84226